Sources/ApodiniHTTP/Configuration.swift

```diff
@@ -0,0 +1,39 @@
+//
+//  Configuration.swift
+//  
+//
+//  Created by Max Obermeier on 30.06.21.
+//
+
+import Foundation
+import Apodini
+import ApodiniExtension
+import ApodiniVaporSupport
+import Vapor
+
+
+extension HTTP {
+    /// Configuration that can be used to customize the behavior of the ``HTTP`` exporter.
+    public struct ExporterConfiguration {
+        /// The `AnyEncoder` to be used for encoding responses
+        public let encoder: AnyEncoder
+        /// The `AnyDecoder` to be used for decoding requests
+        public let decoder: AnyDecoder
+        /// Indicates whether the HTTP route is interpreted case-sensitivly
+        public let caseInsensitiveRouting: Bool
+        
+        
+        /// Initializes the configuration of the ``HTTP`` exporter
+        /// - Parameters:
+        ///    - encoder: The to be used `AnyEncoder`, defaults to a `JSONEncoder`
+        ///    - decoder: The to be used `AnyDecoder`, defaults to a `JSONDecoder`
+        ///    - caseInsensitiveRouting: Indicates whether the HTTP route is interpreted case-sensitivly
+        public init(encoder: AnyEncoder = HTTP.defaultEncoder,
```

Comment:
Just out of curiosity, can the OpenAPI exporter be easily adapted to also depend on the plain http exporter, providing an OpenAPI specification for it as well?

Reply:
That would definitely still require some work, I guess. I didn't want to open that door tbh, but that would definitely be a topic for future work. The implementation of the OpenAPI exporter still pretty "manually" tries to re-build what the REST exporter does at runtime. One would have to establish a generic model at the OpenAPI exporter first.

Reply:
Alright, thanks for explaining. Just was curios.